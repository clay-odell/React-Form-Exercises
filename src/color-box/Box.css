.Box {
    position: relative;
    background-color: var(--background-color);
    width: var(--width);
    height: var(--height);
    display: flex;
    justify-content: center;
    align-items: center;
    margin: 10px;
}

.remove-btn {
    position: absolute;
    top: 5px;
    right: 5px;
    background: red;
    color: white;
    border: none;
    border-radius: 50%;
    width: 20px;
    height: 20px;
    cursor: pointer;
    display: flex;
    justify-content: center;
    align-items: center;
}
