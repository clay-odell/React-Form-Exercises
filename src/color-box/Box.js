import React from 'react';
import './Box.css';

const Box = ({ id, backgroundColor, width, height, removeBox }) => {
    const boxStyle = {
        '--background-color': backgroundColor,
        '--width': `${width}px`,
        '--height': `${height}px`,
    };

    return <div className="Box" style={boxStyle}>
        <button className="remove-btn" onClick={() => removeBox(id)}>X</button>
    </div>
}

export default Box;