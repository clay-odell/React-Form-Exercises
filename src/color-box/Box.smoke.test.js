import React from 'react';
import { render } from '@testing-library/react';
import Box from './Box';

it("renders without crashing", () => {
    render(<Box />);
});

it("matches snapshot", () => {
    const { asFragment } = render(<Box backgroundColor="blue" width="100" height="100" />);
    expect(asFragment()).toMatchSnapshot();
});
