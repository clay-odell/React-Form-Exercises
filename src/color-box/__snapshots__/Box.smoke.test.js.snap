// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
    style="--background-color: blue; --width: 100px; --height: 100px;"
  >
    <button
      class="remove-btn"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
